Behaviour:
  Head:
    Toggles:
      #Wenn zu lange kein Torpfosten gesehen wurde wird kurz auf die Position an der Tropfosten laut LokalGoalModule
      #sein sollte geschaut um die daten zu aktualisieren
      tryRefreshGoalData: False #unused

      #Uses the Visionnformation to focus the goals, if false it will use the information of the GoalModel
      goalVisionTracking: True

      #continuous switching between Ball and Goal searching
      switchBallGoalSearch: False

      # track goal while going on position
      track_goal: True

      # try to find the ball on old position
      look_at_old_position: True

    Camera:
            minPan: -125
            minTilt: -90
            maxPan: 125
            maxTilt: 0

            # The tilt where the robot looks straight forward
            initialTilt: -12

            #this is only right for 4:3 resolution!
            cameraAngle: 66.5

            horizontalFactor: 1
            verticalFactor: 0.5625

            cameraHeight: 0.75
            ballHeight: 0.20

    Tracking:
        #max Speed for tracking (no unit, 0 is max)
        maxPanSpeedTracking: 0
        maxTiltSpeedTracking: 0

        # Maximale abweichung bevor gedreht wird
        xSensivity: 0.1
        # Maximale abweichung bevor gedreht wird
        ySensivity: 0.1

        # Idealer Ballmittelpunkt auf y-achse (-11)
        yCenterDefault: 0

        # Goalie soll das Tracking Center weiter oben im Bild haben
        yCenterGoalie: 0.6

        # Time how long to try to track the Ball after we lost it
        trackBallLost: 0.25

        # counter were
        ballFailCounterMax: 17

    Search:
        #max Speed for searching (no unit, 0 is max)
        maxPanSpeedSearch: 0
        maxTiltSpeedSearch: 0

        # time between steps of the head movement
        headTurnTime: 0.7

        # how exact the head has to be an the position to be recongized as at this position (in degree)
        headTurnPrecision: 2

        # Time in seconds to wait before moving
        turnWaitTime: 5 #todo change if turning works

        #New feature if wew want to use invere kinematics for searching
        useKinematicTaskForSearching: true

        # Time to confirm an object
        confirmTime: 2

        # After this time (in s), the robot will not look next to the last position of a searched object
        objectLostTime: 3

        # When the ball is lost, we are looking at these offsets around its last position
        # the offsets are in meters
        # TODO: evaluate offsets
        offsetRight: 0.30
        offsetDown: 0.30
        offsetLeft: 0.30

    SearchPattern:
      help: |
        These are the search pattern for the head behaviour.
        Each position is a tupel of meters to the front and left.

      ball:
            # TODO: Evaluate patterns
            - [ 0.05,   0]
            - [ 0.3,   0]
            - [ 0.5,   1]
            - [ 0.5,  -1]
            - [ 1.5,-1.5]
            - [ 1.5,   0]
            - [ 1.5, 1.5]
            - [ 2.5,   0]

      goalie:
            - [ 2.5,   0]
            - [ 1.5,   0]
            - [   0, 2.5]
            - [ 1.5, 1.5]
            - [ 1.5,   0]
            - [ 1.5,-1.5]
            - [   0,-2.5]

      defender:
            - [ 2.5,   0]
            - [ 1.5,   0]
            - [   0, 2.5]
            - [ 1.5, 1.5]
            - [ 1.5,   0]
            - [ 1.5,-1.5]
            - [   0,-2.5]

      center:
            - [ 2.5,   0]
            - [ 1.5,   0]
            - [   0, 2.5]
            - [ 1.5, 1.5]
            - [ 1.5,   0]
            - [ 1.5,-1.5]
            - [   0,-2.5]
